clients/shoreline/shoreline.go

```diff
@@ -167,44 +174,81 @@ func (b *ShorelineClientBuilder) Build() *ShorelineClient {
 // Start starts the client and makes it ready for us.  This must be done before using any of the functionality
 // that requires a server token
 func (client *ShorelineClient) Start() error {
-	attempts := 0
-	var err error 
-	for attempts < 5 {
-		if err = client.serverLogin(); err != nil {
-			log.Printf("Problem with initial server token acquisition, [%v]", err)
-			attempts++
-			time.Sleep(10*time.Second) 
-		} else {
-			break 
-		}
-	}
-	
-	if err != nil {
-		return err
+	var err error
+	// for attempts < 5 {
+	if err = client.serverLogin(); err != nil {
+		log.Printf("Error on initial server token acquisition, [%v]", err)
+		go client.serverLoginLoop(false)
+	} else {
+		go client.refreshTokenLoop()
 	}
+	return nil
+}
 
-	go func() {
-		for {
-			timer := time.After(time.Duration(client.config.TokenRefreshInterval))
-			select {
-			case twoWay := <-client.closed:
-				twoWay <- true
+func (client *ShorelineClient) serverLoginLoop(refresh bool) {
+	var attempts int64
+	client.mut.Lock()
+	if client.acquiringToken {
+		client.mut.Unlock()
+		return
+	}
+	client.acquiringToken = true
+	client.mut.Unlock()
+	for {
+		timer := time.After(time.Duration(client.config.TokenGetInterval))
+		select {
+		case twoWay := <-client.closed:
+			twoWay <- true
+			return
+		case <-timer:
+			err := client.serverLogin()
+			if err == nil {
+				log.Printf("Server token acquired successfully after %v attempts", attempts)
+				client.mut.Lock()
+				client.acquiringToken = false
+				client.mut.Unlock()
+				if !refresh {
+					go client.refreshTokenLoop()
+				}
```

Comment:
this part looks strange `if not refresh do refresh`. Not sure to understand it

Reply:
`refresh` parameter may be renamed to `launchRefreshTokenLoop`  (and invert the boolean set in routine calls) ?

Reply:
would make sense to ease the readability ;-)